src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -305,28 +305,34 @@ private void evaluateSystemIndicesAccess(
         // cluster actions will return true for requestedResolved.isLocalAll()
         // the following section should only be run for index actions
         if (this.isSystemIndexEnabled && user.isPluginUser() && !requestedResolved.isLocalAll()) {
-            Set<String> matchingSystemIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
+            Set<String> matchingPluginIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
                 user.getName().replace("plugin:", ""),
                 requestedResolved.getAllIndices()
             );
-            if (requestedResolved.getAllIndices().equals(matchingSystemIndices)) {
+            if (requestedResolved.getAllIndices().equals(matchingPluginIndices)) {
                 // plugin is authorized to perform any actions on its own registered system indices
                 presponse.allowed = true;
                 presponse.markComplete();
+                return;
             } else {
-                if (log.isInfoEnabled()) {
-                    log.info(
-                        "Plugin {} can only perform {} on it's own registered System Indices. System indices from request that match plugin's registered system indices: {}",
-                        user.getName(),
-                        action,
-                        matchingSystemIndices
-                    );
+                Set<String> matchingSystemIndices = SystemIndexRegistry.matchesSystemIndexPattern(requestedResolved.getAllIndices());
+                matchingSystemIndices.removeAll(matchingPluginIndices);
+                // See if request matches other system indices not belong to the plugin
+                if (!matchingSystemIndices.isEmpty()) {
+                    if (log.isInfoEnabled()) {
+                        log.info(
+                            "Plugin {} can only perform {} on it's own registered System Indices. System indices from request that match plugin's registered system indices: {}",
+                            user.getName(),
+                            action,
+                            matchingPluginIndices
+                        );
+                    }
+                    presponse.allowed = false;
```

Comment:
Do I understant it right that the goal of this code is to deny access to other plugin indices even if the particular plugin has listed it in the plugin-permissions.yml file?